src/gradata/hooks/secret_scan.py

```
@@ -0,0 +1,76 @@
+    ("db_conn_string",   re.compile(r"(?:postgres|mysql|mongodb|redis)://[^:]+:[^@]+@[^\s\"']+", re.I)),
+    ("generic_secret",   re.compile(r"(?:password|api_key|token|secret|apikey|api_secret)\s*[=:]\s*[\"']?[^\s\"']{8,}", re.I)),
+]
+
+
+def _scan_content(content: str) -> list[dict]:
+    """Scan a string for secret patterns. Returns list of findings."""
+    findings = []
+    for name, pattern in SECRET_PATTERNS:
+        matches = pattern.findall(content)
+        if matches:
+            for m in matches:
```

Ruff (0.15.9)

[warning] Loop control variable `m` not used within loop body

Rename unused `m` to `_m`

(B007)